diff --git a/node_modules/@analogjs/astro-angular/src/index.js b/node_modules/@analogjs/astro-angular/src/index.js
index 277cc8d..0f85d4a 100644
--- a/node_modules/@analogjs/astro-angular/src/index.js
+++ b/node_modules/@analogjs/astro-angular/src/index.js
@@ -15,7 +15,7 @@ function getRenderer(ngHydration) {
 function getViteConfiguration(options) {
     return {
         esbuild: {
-            jsxDev: true,
+            jsxDev: false,
         },
         optimizeDeps: {
             include: [
