diff --git a/node_modules/@qwik.dev/astro/src/plugins.ts b/node_modules/@qwik.dev/astro/src/plugins.ts
index f07cd9b..97e90e3 100644
--- a/node_modules/@qwik.dev/astro/src/plugins.ts
+++ b/node_modules/@qwik.dev/astro/src/plugins.ts
@@ -13,7 +13,7 @@ import { SERVER_ENTRYPOINT } from "./constants";
  */
 export function stripClientOutputHooks(plugins: PluginOption[]) {
   for (const plugin of plugins) {
-    if (!plugin || typeof plugin !== "object") continue;
+    if (!plugin || typeof plugin !== "object" || !("name" in plugin)) continue;
     if ("outputOptions" in plugin) delete plugin.outputOptions;
     if (plugin.name === "vite-plugin-qwik-post" && "generateBundle" in plugin)
       delete plugin.generateBundle;
